sw/device/lib/dif/README.md

```diff
@@ -105,63 +125,50 @@ extracted from the IP's auto-generated register header file, e.g.,
 
 #### Base Types
 
-The following basic types are expected to be provided by all DIFs (unless
-otherwise specified).
+There are two categories of base types:
+1. those that are defined once in `sw/device/lib/dif/dif_base.h` and used in all
+   DIF libraries, and
+2. those that are expected to be defined separately by all DIF libraries (unless
+   otherwise specified).
+
+The base types defined in `sw/device/lib/dif/dif_base.h` include:
+* `dif_result_t` -- an enum representing global DIF return codes.
+* `dif_toggle_t` -- an enum to be used instead of a `bool` when describing the
+  enablement state of a CSR.
```

Comment:
Not sure if this reads ok but `dif_toggle_t` is not really about CSRs, right?

```suggestion
* `dif_toggle_t` -- an enum to be used instead of a `bool` when describing an
  enablement state.
```

Reply:
Fixed.